### PR TITLE
Allow df.check.nunique() to accept multiple columns

### PR DESCRIPTION
**Pull request by cparmet**
Partially addresses #60 

For Dataframe `check.nunique()`:
* Change `column` argument to `columns`
* When `columns` is a str, behavior is unchanged: run Pandas `nunique()` on that single Series
* When `columns` is a list, count the number of unique rows considering the subset of `columns`
* When `columns` is None (default), consider the number of unique rows considering all columns 

Pull request checklist
- [X] PR describes the changes proposed
- [X] Tests were checked for updates
- [X] Tests pass with `nox`
- [X] Docstrings and docs were checked for updates

### CODE REVIEW

**Comment by cparmet** [comment]
@alexblakes Can you give this PR a test drive?

File: pandas_checks/DataFrameChecks.py [hunk 1]
```diff
@@ -1474,25 +1474,25 @@ def nrows(
 
     def nunique(
         self,
-        column: str,
+        columns: Union[str, list, None] = None,
```
Comment (alexblakes) [inline pandas_checks/DataFrameChecks.py]: Should we explicitly check that column is str, list, or None? 

I tried passing other iterables e.g. pd.Series (ValueError), tuple (works), set (TypeError) with mixed results.

**Review by alexblakes — COMMENTED** [review]
Two tiny comments and it otherwise looks fab!

**Comment by alexblakes** [comment]
Oh wait, no. I am getting a RecursionError when calling `nunique()` without arguments.

File: pandas_checks/DataFrameChecks.py [hunk 2]
```diff
@@ -1502,15 +1502,33 @@ def nunique(
...
-                _apply_modifications(self._obj, fn=fn, subset=column)
-                .check.nunique(
-                    fn=lambda df: df,  # Identity function
-                    check_name=check_name,
-                    **kwargs,
-                )
-            )  # fmt: skip
+            # Ensure columns_clean is a list
+            columns_clean = (
+                [columns] if (isinstance(columns, str) or not columns) else columns
```
Comment (cparmet) [inline pandas_checks/DataFrameChecks.py]: Good call!

File: pandas_checks/DataFrameChecks.py (hunk 1 above)
Comment (cparmet) [reply pandas_checks/DataFrameChecks.py]: Nice idea

File: pandas_checks/DataFrameChecks.py [hunk 3]
```diff
@@ -1502,15 +1502,33 @@ def nunique(
...
+            )
+            if len(columns_clean) == 0:
+                # When columns=None, consider all columns
+                columns_clean = self._obj.columns.tolist()
```
Comment (alexblakes) [inline pandas_checks/DataFrameChecks.py]: I ran into some trouble with:
- Calling `check.nunique()` without arguments raised a RecursionError
- Passing an empty string `""` or a list of one empty string `[""]` were treated as `None` (I would expect a KeyError, i.e. none of the columns are named `""`)
- Other Falsey arguments were treated as `None` - but `0` is a valid (any just about plausible) column name.

Here's what I came up with to address those issues... Feel free to reject or refactor for consistency with the rest of the library.

```suggestion
        if get_mode()["enable_checks"]:

            if not isinstance(columns, str | int | list | None):
                raise ValueError("'columns' must be a string, an int, a list, or None")
            if columns is None:
                columns_clean = self._obj.columns.tolist()
            elif columns == 0:
                columns_clean = [columns]
            elif not columns:
                raise ValueError("'columns' cannot be an empty list or empty string")
            else:
                columns_clean = [columns]
```

**Comment by cparmet** [comment]
Wow @alexblakes thanks for the great tests and catching these issues, and working out solutions! Sorry that I didn't your suggested changes till I had been working through my own version -- I was arm wrestling with an edge case in the test suite with MultIindex columns and a few `mypy` type barks.

I went back and I think I've integrated your changes and we come to almost the same place?

In the latest commits, it allows `columns` to be:
* str
* list
* `None`, without an infinite recursion hole!
* 0 or other int
* False
* ""

(assuming these "columns" are in the DF)

I think the last item is the difference, as this commit allows `columns = ""`, even if I can't immediately imagine that scenario :D

What do you think?

**Comment by cparmet** [comment]
@alexblakes Sorry hold please, I realized an issue, my new code ignores the common dropna argument

**Comment by cparmet** [comment]
@alexblakes My apologies, I realized a couple problems with the solution I originally proposed in #60 ! I've started over in a new simpler PR #67

Problems with my approach
1. Pandas already has an .nunique() method for DataFrames. It was not accessible from Pandas Checks. That should be fixed!
2. The Pandas method counts the unique values in _each column_ separately. So if we proceeded with my original approach, the `check.nunique()` with a `columns` parameter would (A) differ from the Pandas API, and (B) the default behavior of nunique() would conflict, which is confusing.
3. In Pandas the Series and DataFrame .nunique() methods both default to `dropna=True`. The approach I had above would not consider dropna by default for DataFrames, which is tricky to generalize in our 'across columns' scenario, and again confusing given the conflicting APIs

I'll create a separate PR to fix the other issues you identified that occur when column name (`subset`) is 0, "", or False. These are bugs in Pandas Checks generally, potentially applying to all methods with `subset` (which the new approach to `nunique()` simply taps into). So thank you for identifying those!

I'll close this PR for now, and see ya over in the new one, but if I'm thinking about this poorly (again!) we can always reincarnate this PR!
